kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/InBandKeyManagerTest.java

```diff
@@ -181,8 +185,89 @@ void encryptionRetry() {
         CompletionStage encrypt = km.encrypt("topic", 1, new EncryptionScheme<>(kekId, EnumSet.of(RecordField.RECORD_VALUE)),
                 initial,
                 recordReceivedRecord(encrypted));
-        String expectedMessage = "failed to encrypt records for topic topic partition 1 after 3 attempts";
-        assertThat(encrypt).failsWithin(Duration.ZERO).withThrowableThat().withMessageContaining(expectedMessage);
+        assertThat(encrypt).failsWithin(Duration.ofSeconds(5)).withThrowableThat()
+                .withMessageContaining("failed to encrypt records for topic topic partition 1 after 3 attempts");
+    }
+
+    @Test
+    void dekCreationRetryFailurePropagatedToEncryptCompletionStage() {
+        var kmsService = UnitTestingKmsService.newInstance();
+        InMemoryKms kms = kmsService.buildKms(new UnitTestingKmsService.Config());
+        var kekId = kms.generateKey();
+        InMemoryKms spyKms = Mockito.spy(kms);
+        when(spyKms.generateDekPair(kekId)).thenReturn(CompletableFuture.failedFuture(new EncryptorCreationException("failed to create that DEK")));
```

Comment:
Feels odd to spy on a test KMS instance that could easily be configured to return failed futures for a specific `kekId`